packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -165,12 +209,82 @@ export class PendingStateManager {
         this.pendingStates.push(pendingFlush);
     }
 
+    public rebaseAt(seqNum: number) {
+        const rebasePromises: Promise<void>[] = [];
+        // rebase initial ops at sequence number
+        while (!this.initialStates.isEmpty()) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const nextState = this.initialStates.peekFront()!;
+            if (nextState.type === "message") {
+                if (nextState.referenceSequenceNumber > seqNum) {
+                    break; // nothing left to do at this sequence number
+                } else if (nextState.referenceSequenceNumber > 0 && nextState.referenceSequenceNumber < seqNum) {
+                    // this method should be called at every sequence number in order, so this should never happen
+                    if (nextState.clientSequenceNumber === this.initialClientSeqNum) {
+                        throw new Error("loaded from snapshot too recent to rebase pending initial ops");
+                    } else {
+                        throw new Error("gap in message sequence or messages out of order");
+                    }
+                }
+
+                // rebaseOp will cause the DDS to behave as if it has sent the op but not actually send it
+                rebasePromises.push(this.rebaseOp(nextState.content, nextState.localOpMetadata));
```

Comment:
I do not think this is correct. I believe correct behavior here is to await.
In other words, I'm not convinced that it's safe to allow processing of two local ops in parallel.
Code also becomes simpler and easier to read